add.go

```diff
@@ -188,10 +188,39 @@ func includeDirectoryAnyway(path string, pm *fileutils.PatternMatcher) bool {
 	return false
 }
 
+// validateSourceAndDest performs a sanity check on the case that if more
+// than one source is provided then destination must end with a slash (`/`),
+// which signifies that destination is a directory and similarly function
+// returns error if destination does not ends with a slash (`/`).
+// Following function can contain more validation cases in future.
+func validateSourceAndDest(destination string, sources []string) error {
+	multipleSources := false
+	if len(sources) > 1 {
+		multipleSources = true
+	} else {
+		// Check length before accessing `0`
+		// to avoid `index out of range`.
+		if len(sources) == 1 {
+			if strings.HasSuffix(sources[0], "*") {
+				multipleSources = true
+			}
+		}
+	}
+	if multipleSources {
+		if !strings.HasSuffix(destination, "/") {
```

Comment:
Does docker enforce this?  Or does it just check if destination is not a directory then throw an error?

Reply:
Just chiming in that if the directory already exists, podman/buildah does the expected (?) thing and copies all the matches of the wildcard into the existing directory

Re-using the reproducer from #4167, if you modify the Dockerfile to this
```dockerfile
FROM alpine:latest as builder

RUN mkdir /spam
COPY spam/*.txt /spam
```
Then it copies all the files

Reply:
I think docker checks this cause rule is
```
* If multiple <src> resources are specified, either directly or due to the use of a wildcard, then <dest> must be a directory, and it must end with a slash /.
```
https://docs.docker.com/engine/reference/builder/#add

Reply:
I still believe this should be done on in the copier.

Basically if the SRC contains more then one file and the DEST is not a directory, then it should return an error.

The directory should not be checked for ending with /

Reply:
We're already globbing inside a chroot further down, when we call `copier.Stat()`, and we're probably better off doing this sort of check just after that, when we have all of the facts.  Doing this before that means that the current iteration of this patch passes URLs for remote locations to `filepath.Glob()`, which is a bug.

Reply:
I guess they do document the following:
```
If <dest> does not end with a trailing slash, it will be considered a regular file and the contents of <src> will be written at <dest>.
```

But @nalind question above has not been addressed

Reply:
> Basically if the SRC contains more then one file and the DEST is not a directory, then it should return an error.
>
> The directory should not be checked for ending with /

+1 to this, it sounds reasonable to me that if the destination already exists then the trailing slash doesn't need to be enforced. It wouldn't be 100% consistent with docker, but isn't it likely that some users are already doing this? E.g. copying things to `/etc` or other standard dirs

Reply:
@chmeliik Yes that is already addressed I just need to move code around as per @nalind 's suggestion here https://github.com/containers/buildah/pull/4183#discussion_r953873090 , I'll amend this.

Reply:
Done moved to just after we split local and remote sources and only validating local sources now.